src/NUnitFramework/framework/Api/FrameworkController.cs

```diff
@@ -385,18 +385,14 @@ private static void AddSetting(TNode settingsNode, string name, object value)
             TNode setting = new TNode("setting");
             setting.AddAttribute("name", name);
 
-            if (value != null)
+            if (value is IDictionary dict)
```

Comment:
The only not wholly trivial change here. Since this was inside a null check but `is` also performs a null check I removed the outer null check to reduce nesting.